hazelcast/src/main/java/com/hazelcast/client/impl/connection/tcp/TcpClientConnectionManager.java

```diff
@@ -1074,6 +1046,42 @@ private TcpClientConnection onAuthenticated(TcpClientConnection connection,
         return connection;
     }
 
+    private void connectToTpcPorts(TcpClientConnection connection, List<Integer> tpcPorts) throws IOException {
+        Channel[] tpcChannels = new Channel[tpcPorts.size()];
+        for (int k = 0; k < tpcChannels.length; k++) {
+            Address address = new Address(connection.getRemoteAddress().getHost(), tpcPorts.get(k));
+            logger.info("TPC Client: Connecting to:" + address);
+            SocketChannel tpcSocketChannel = SocketChannel.open();
+
+            Channel tpcChannel = networking.register(connection.getChannelInitializer(), tpcSocketChannel, true);
+            tpcChannel.attributeMap().put(Address.class, address);
+            tpcChannel.attributeMap().put(TcpClientConnection.class, connection);
+
+            InetSocketAddress tpcSocketChannelAddress = new InetSocketAddress(address.getHost(), address.getPort());
+
+            tpcChannel.connect(tpcSocketChannelAddress, connectionTimeoutMillis);
+
+            logger.info("TPC Client: Successfully connected to " + tpcSocketChannelAddress);
+
+            tpcSocketChannel.configureBlocking(false);
+
+            tpcChannel.start();
+
+            // first thing we need to send is the clientUuid so this new socket can be connected
+            // to the connection on the member.
+            ClientMessage clientUuidMessage = ClientMessage.createForEncode();
+            ClientMessage.Frame initialFrame = new ClientMessage.Frame(
+                    new byte[BOOLEAN_SIZE_IN_BYTES + 2 * LONG_SIZE_IN_BYTES], UNFRAGMENTED_MESSAGE);
+            encodeUUID(initialFrame.content, 0, clientUuid);
+            clientUuidMessage.add(initialFrame);
```

Comment:
Can you please have this code generated via the protocol generator?

Reply:
As we talked, these are from the PoC implementation and will be changed in future PRs. This doesn't even follow the structure of the client message anyways. Will it be OK for you to leave it like this?